### PR TITLE
UniversalMediaEngine Release 1.1.0.3

### PR DESCRIPTION
**Pull request by bethoma**
Notes:
	- Add Unit Tests

### CODE REVIEW

File: UniversalMediaEngine.Tests/MediaEngineTests.cs [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+            // Act
+            var result = await engineUnderTest.InitializeAsync();
+
+            engineUnderTest.MediaStateChanged += ((MediaState newState) => 
+            {
+                if (MediaState.Playing == newState)
+                    callbackOccurred = true;
+            });
+
+            Assert.IsTrue(MediaEngineInitializationResult.Success == result);
```
Comment (mwmckee) [inline UniversalMediaEngine.Tests/MediaEngineTests.cs]: should this assert move up a few lines to right under where "result" is set?

File: UniversalMediaEngine.Tests/MediaEngineTests.cs [hunk 2]
```diff
@@ -0,0 +1,533 @@
...
+            try
+            {
+                engineUnderTest.Play("ms-appx:///Assets/TestSilence.wav");
+
+                // Wait for the callback to be invoked until the timeout expires
+                uint counter = 0;
+                while (!callbackOccurred)
+                {
+                    if (counter == callbackTimeout)
+                        Assert.IsFalse(true, "Timed out waiting for callback from mediaengine");
```
Comment (mwmckee) [inline UniversalMediaEngine.Tests/MediaEngineTests.cs]: Might this be cleaner as:

Assert.IsFalse(counter == callbackTimeout, "timed out...")

File: UniversalMediaEngine.Tests/MediaEngineTests.cs [hunk 3]
```diff
@@ -0,0 +1,533 @@
...
+            // Act
+            try
+            {
+                engineUnderTest.Play("");
+            }
+            catch(Exception e)
+            {
+                exceptionThrown = true;
+            }
+
```
Comment (mwmckee) [inline UniversalMediaEngine.Tests/MediaEngineTests.cs]: Consistency: missing "//assert" comment

File: UniversalMediaEngine.Tests/MediaEngineTests.cs [hunk 4]
```diff
@@ -0,0 +1,533 @@
...
+            engineUnderTest.MediaStateChanged += ((MediaState newState) =>
+            {
+                // We're looking for the error state as this should be returns for
+                // an invalid URL
+                if (MediaState.Error == newState)
+                    callbackOccurred = true;
+            });
+
+            Assert.IsTrue(MediaEngineInitializationResult.Success == result);
```
Comment (mwmckee) [inline UniversalMediaEngine.Tests/MediaEngineTests.cs]: ordering, consider moving this up under where "result" is set (same comment as my earlier one)

File: UniversalMediaEngine.Tests/MediaEngineTests.cs [hunk 5]
```diff
@@ -0,0 +1,533 @@
...
+            try
+            {
+                engineUnderTest.Play("");
+
+                // Wait for the callback to be invoked until the timeout expires
+                uint counter = 0;
+                while (!callbackOccurred)
+                {
+                    if (counter == callbackTimeout)
+                        Assert.IsFalse(true, "Timed out waiting for callback from mediaengine");
```
Comment (mwmckee) [inline UniversalMediaEngine.Tests/MediaEngineTests.cs]: Might this be cleaner as:

Assert.IsFalse(counter == callbackTimeout, "timed out...")

(same as earlier comment)

File: UniversalMediaEngine.Tests/MediaEngineTests.cs [hunk 6]
```diff
@@ -0,0 +1,533 @@
...
+            try
+            {         
+                // Open the file as a byte stream
+                engineUnderTest.PlayStream(await testFile.OpenReadAsync());
+
+                // Wait for the callback to be invoked until the timeout expires
+                uint counter = 0;
+                while (!callbackOccurred)
+                {
+                    if (counter == callbackTimeout)
```
Comment (mwmckee) [inline UniversalMediaEngine.Tests/MediaEngineTests.cs]: Might this be cleaner as:

Assert.IsFalse(counter == callbackTimeout, "timed out...")

(same as earlier comment)

File: UniversalMediaEngine.Tests/MediaEngineTests.cs [hunk 7]
```diff
@@ -0,0 +1,533 @@
...
+                            if (MediaState.Playing == newState)
+                            {
+                                secondPlayingCallbackOccurred = true;
+                                callbackCount++;
+                            }
+                            break;
+                }
+            });
+
+            Assert.IsTrue(MediaEngineInitializationResult.Success == result);
```
Comment (mwmckee) [inline UniversalMediaEngine.Tests/MediaEngineTests.cs]: consider moving up. Same as earlier comments.

File: UniversalMediaEngine.Tests/Package.appxmanifest [hunk 8]
```diff
@@ -0,0 +1,45 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Package
+  xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10"
+  xmlns:mp="http://schemas.microsoft.com/appx/2014/phone/manifest"
+  xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10"
+  IgnorableNamespaces="uap mp">
+
+  <Identity Name="0dec6c77-5b50-49a9-9d49-1186904b5df9"
+            Publisher="CN=bethoma"
```
Comment (mwmckee) [inline UniversalMediaEngine.Tests/Package.appxmanifest]: Microsoft?

File: UniversalMediaEngine/MediaEngine.cpp [hunk 9]
```diff
@@ -6,6 +6,11 @@ using namespace Microsoft::Maker::Media::UniversalMediaEngine;
 using namespace Platform;
 using namespace concurrency;
 
+MediaEngine::MediaEngine()
```
Comment (mwmckee) [inline UniversalMediaEngine/MediaEngine.cpp]: why the change to move where this is created?

File: UniversalMediaEngine.Tests/UniversalMediaEngine.Tests.csproj [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+    <RootNamespace>UniversalMediaEngine.Tests</RootNamespace>
+    <AssemblyName>UniversalMediaEngine.Tests</AssemblyName>
+    <DefaultLanguage>en-US</DefaultLanguage>
+    <TargetPlatformIdentifier>UAP</TargetPlatformIdentifier>
+    <TargetPlatformVersion>10.0.10586.0</TargetPlatformVersion>
+    <TargetPlatformMinVersion>10.0.10240.0</TargetPlatformMinVersion>
+    <MinimumVisualStudioVersion>14</MinimumVisualStudioVersion>
+    <FileAlignment>512</FileAlignment>
+    <ProjectTypeGuids>{A5A43C5B-DE2A-4C0C-9213-0A381AF9435A};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
+    <PackageCertificateKeyFile>UniversalMediaEngine.Tests_TemporaryKey.pfx</PackageCertificateKeyFile>
```
Comment (mwmckee) [inline UniversalMediaEngine.Tests/UniversalMediaEngine.Tests.csproj]: should remove .pfx file and reference

**Comment by mwmckee** [comment]
please add the project.lock.json files to the .gitignore

File: UniversalMediaEngine/MediaEngine.cpp (hunk 9 above)
Comment (bethoma) [inline UniversalMediaEngine/MediaEngine.cpp]: To fix a bug that caused an AV when methods were called pre init

File: .gitignore [hunk 11]
```diff
@@ -6,6 +6,7 @@
 *.user
 *.userosscache
 *.sln.docstates
+*.project.lock
```
Comment (mwmckee) [inline .gitignore]: need to append with ".json". I think 14.8k of your 15.8 k lines of changes are just that one file, and its still here :P

File: samples/basicMediaEngineSample/basicMediaEngineSample/MainPage.xaml.cpp [hunk 12]
```diff
@@ -37,7 +37,7 @@ void basicMediaEngineSample::MainPage::Page_Loaded(Platform::Object^ sender, Win
 
 void basicMediaEngineSample::MainPage::Button_Click(Platform::Object^ sender, Windows::UI::Xaml::RoutedEventArgs^ e)
 {
-	m_mediaEngine.Play(UrlBox->Text);
+	//m_mediaEngine.Play(UrlBox->Text);
```
Comment (ooeygui) [inline samples/basicMediaEngineSample/basicMediaEngineSample/MainPage.xaml.cpp]: Commented out code? Comment when this should be reenabled, or remove

File: samples/speechSynthesisSample/speechSynthesisSample/Package.appxmanifest [hunk 13]
```diff
@@ -8,14 +8,30 @@
 
   <Identity
     Name="4474c45b-baf1-4ef5-ae2c-962ca5eafb10"
+<<<<<<< HEAD
```
Comment (ooeygui) [inline samples/speechSynthesisSample/speechSynthesisSample/Package.appxmanifest]: Did you check a merge in?

**Comment by ooeygui** [comment]
Looks like several merge conflicts were committed

**Comment by ooeygui** [comment]
:+1:

**Comment by mwmckee** [comment]
Signoff
